viz_scripts/naive_cheer_comparison.ipynb

```diff
@@ -0,0 +1,2200 @@
+    "\n",
+    "    # ensure all modes, including \"Walking\" and \"Bicycling\", are represented\n",
+    "    for mode in ['Walking', 'Bicycling']:\n",
+    "        if mode not in cumulative_co2_emission['predicted_mode_name'].values:\n",
+    "            cumulative_co2_emission = pd.concat([\n",
+    "                cumulative_co2_emission,\n",
+    "                pd.DataFrame({'predicted_mode_name': [mode], 'cumulative_co2_emission_kg_naive_naive': [0]})\n",
+    "            ], ignore_index=True)\n",
+    "\n",
+    "    # exclude specific modes\n",
+    "    cumulative_co2_emission = cumulative_co2_emission[~cumulative_co2_emission['predicted_mode_name'].isin(['Tram', 'Light_Rail', 'Unknown'])]\n",
+    "\n",
+    "    #\n",
+    "    # sensed mode to data user input confusion matrix\n",
+    "    #\n",
+    "        #  temporary merged DataFrame for the confusion matrix\n",
+    "\n",
+    "    temp_joined_df = pd.merge(\n",
+    "        dataframe,\n",
+    "        full_dataframe[['data_cleaned_trip', 'data_user_input_mode_confirm']],\n",
+    "        left_on=trip_id_column,\n",
+    "        right_on='data_cleaned_trip',\n",
+    "        how='left'\n",
+    "    )\n",
+    "    \n",
+    "    comparison_table = pd.crosstab(\n",
+    "        temp_joined_df['data_user_input_mode_confirm'], \n",
+    "        temp_joined_df['predicted_mode_name'], \n",
+    "        rownames=['User Input Mode'], \n",
+    "        colnames=['Predicted Mode'],\n",
+    "        normalize='index'  # Normalize by rows to show percentages\n",
+    "    ) * 100\n",
+    "\n",
+    "    # output the raw comparison matrix and normalized version\n",
+    "    raw_comparison_table = pd.crosstab(\n",
+    "        temp_joined_df['data_user_input_mode_confirm'], \n",
+    "        temp_joined_df['predicted_mode_name'], \n",
+    "        rownames=['User Input Mode'], \n",
+    "        colnames=['Predicted Mode']\n",
+    "    )\n",
+    "\n",
+    "    print(\"Raw Comparison Table:\")\n",
+    "    print(raw_comparison_table)\n",
+    "\n",
+    "    print(\"\\nNormalized Comparison Table (%):\")\n",
+    "    print(comparison_table)\n",
+    "    \n",
+    "    #\n",
+    "    #\n",
+    "    \n",
+    "    \n",
+    "    # filter out modes with zero cumulative emissions\n",
+    "    cumulative_co2_emission_filtered = cumulative_co2_emission[\n",
+    "        cumulative_co2_emission['cumulative_co2_emission_kg_naive_naive'] > 0\n",
+    "    ]\n",
+    "    # calculate cumulative distance for each mode\n",
+    "    cumulative_distance = dataframe.groupby('predicted_mode_name')['distance_km'].sum().reset_index()\n",
+    "\n",
+    "    # rename the column to reflect cumulative distance\n",
+    "    cumulative_distance.columns = ['predicted_mode_name', 'cumulative_distance_km']\n",
+    "\n",
+    "    # filter out modes with absolutely zero distance and exclude Tram and Unknown\n",
+    "    cumulative_distance_filtered = cumulative_distance[\n",
```

Comment:
Is this section for the distane by sense mode chart you had made? Can you label it so it is more clear what it is